native/src/main/java/io/ballerina/lib/aws/mpe/AwsMpeClient.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.lib.aws.mpe;
+
+import io.ballerina.runtime.api.Environment;
+import io.ballerina.runtime.api.Future;
+import io.ballerina.runtime.api.values.BError;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BObject;
+import io.ballerina.runtime.api.values.BString;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.AwsSessionCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.services.marketplaceentitlement.MarketplaceEntitlementClient;
+import software.amazon.awssdk.services.marketplaceentitlement.model.GetEntitlementsRequest;
+import software.amazon.awssdk.services.marketplaceentitlement.model.GetEntitlementsResponse;
+
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * Representation of {@link software.amazon.awssdk.services.marketplaceentitlement.MarketplaceEntitlementClient} with
+ * utility methods to invoke as inter-op functions.
+ */
+public final class AwsMpeClient {
```

Comment:
+1 for the suggestion. Rather than `Client` what do you think about a name like `ClientNativeHandler` or `ClientNativeAdaptor` as this class is not exactly the Java client we are invoking ?

Reply:
Yes +1. or we can mark it as `NativeClient` to avoid the confusion you mentioned.

Reply:
Agreed. Will proceed with `NativeClientAdaptor`.

Reply:
Fixed with [e752f6e3a191a1c527bdcb5e714ee15d34903469](https://github.com/ballerina-platform/module-ballerinax-aws.marketplace.mpe/pull/2/commits/e752f6e3a191a1c527bdcb5e714ee15d34903469)